aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -549,12 +594,19 @@ module aptos_framework::fungible_asset {
     fun withdraw_internal(
         store_addr: address,
         amount: u64,
-    ): FungibleAsset acquires FungibleStore {
+    ): FungibleAsset acquires FungibleStore, ConcurrentFungibleBalance {
         assert!(amount != 0, error::invalid_argument(EAMOUNT_CANNOT_BE_ZERO));
-        let store = borrow_global_mut<FungibleStore>(store_addr);
-        assert!(store.balance >= amount, error::invalid_argument(EINSUFFICIENT_BALANCE));
-        store.balance = store.balance - amount;
 
+        if (concurrent_fungible_balance_exists(store_addr)) {
+            let balance = borrow_global_mut<ConcurrentFungibleBalance>(store_addr);
+            assert!(aggregator_v2::try_sub(&mut balance.balance, amount), error::invalid_argument(EINSUFFICIENT_BALANCE));
+        } else {
+            let store = borrow_global_mut<FungibleStore>(store_addr);
```

Comment:
We don't check store exists here, is the check somewhere else?

Reply:
yes. delegate to caller